descriptor/qsarmolecular/src/main/java/org/openscience/cdk/qsar/descriptors/molecular/LongestAliphaticChainDescriptor.java

```diff
@@ -18,71 +18,67 @@
  */
 package org.openscience.cdk.qsar.descriptors.molecular;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import org.openscience.cdk.CDKConstants;
 import org.openscience.cdk.exception.CDKException;
-import org.openscience.cdk.exception.NoSuchAtomException;
-import org.openscience.cdk.graph.PathTools;
-import org.openscience.cdk.graph.SpanningTree;
-import org.openscience.cdk.graph.matrix.ConnectionMatrix;
+import org.openscience.cdk.graph.Cycles;
+import org.openscience.cdk.graph.GraphUtil;
 import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 import org.openscience.cdk.interfaces.IBond;
-import org.openscience.cdk.interfaces.IRingSet;
 import org.openscience.cdk.qsar.AbstractMolecularDescriptor;
 import org.openscience.cdk.qsar.DescriptorSpecification;
 import org.openscience.cdk.qsar.DescriptorValue;
-import org.openscience.cdk.qsar.IMolecularDescriptor;
 import org.openscience.cdk.qsar.result.IDescriptorResult;
 import org.openscience.cdk.qsar.result.IntegerResult;
-import org.openscience.cdk.tools.manipulator.AtomContainerManipulator;
 
 /**
- *  Class that returns the number of atoms in the longest aliphatic chain.
- *
+ * Counts the number of atoms in the longest aliphatic chain.
+ * <p>
  * <table border="1"><caption>Parameters for this descriptor:</caption>
- *   <tr>
- *     <td>Name</td>
- *     <td>Default</td>
- *     <td>Description</td>
- *   </tr>
- *   <tr>
- *     <td>checkRingSystem</td>
- *     <td>false</td>
- *     <td>True is the CDKConstant.ISINRING has to be set</td>
- *   </tr>
+ * <tr>
+ * <td>Name</td>
+ * <td>Default</td>
+ * <td>Description</td>
+ * </tr>
+ * <tr>
+ * <td>checkRingSystem</td>
+ * <td>false</td>
+ * <td>True is the CDKConstant.ISINRING has to be set</td>
+ * </tr>
  * </table>
- *
+ * <p>
  * Returns a single value named <i>nAtomLAC</i>
- * @author      chhoppe from EUROSCREEN
+ *
+ * @author chhoppe from EUROSCREEN
+ * @author John Mayfield
  * @cdk.created 2006-1-03
- * @cdk.module  qsarmolecular
+ * @cdk.module qsarmolecular
  * @cdk.githash
  * @cdk.dictref qsar-descriptors:largestAliphaticChain
  */
-public class LongestAliphaticChainDescriptor extends AbstractMolecularDescriptor implements IMolecularDescriptor {
+public class LongestAliphaticChainDescriptor
+    extends AbstractMolecularDescriptor {
```

Comment:
Ah, does it make this class also "implement" what the abstract class implements? I did not know that...

Reply:
Of course